lua/lspsaga/utils.lua

```diff
@@ -0,0 +1,28 @@
+local M = {}
+
+function M.as_table(value)
+  return type(value) == 'string' and { value } or value
+end
+
+--- Creates a buffer local mapping.
+---@param buffer number
+---@param modes string|table<string>
+---@param keys string|table<string>
+---@param action string|function
+---@param opts? table
+function M.map_keys(buffer, modes, keys, action, opts)
```

Comment:
mode is better in opts.mode then  we can do `local mode = opts.mode or 'n'`  . `action` change to `rhs` .

Reply:
If we add `mode` to `opts` then we would need to remove this field before we use it in `nvim_buf_set_keymap` since `opts.mode` is not valid.